ggml/src/ggml-cpu/ggml-cpu.c

```diff
@@ -2417,12 +2417,30 @@ static bool ggml_thread_apply_priority(int32_t prio) {
     // This is up to the applications.
     DWORD p = THREAD_PRIORITY_NORMAL;
     switch (prio) {
+        case GGML_SCHED_PRIO_LOW:      p = THREAD_PRIORITY_BELOW_NORMAL;  break;
         case GGML_SCHED_PRIO_NORMAL:   p = THREAD_PRIORITY_NORMAL;        break;
         case GGML_SCHED_PRIO_MEDIUM:   p = THREAD_PRIORITY_ABOVE_NORMAL;  break;
         case GGML_SCHED_PRIO_HIGH:     p = THREAD_PRIORITY_HIGHEST;       break;
         case GGML_SCHED_PRIO_REALTIME: p = THREAD_PRIORITY_TIME_CRITICAL; break;
     }
 
+    if (prio != GGML_SCHED_PRIO_LOW) {
+        // Tell Windows that this thread should not be throttled (needs its own CPU core).
+        // Newer Windows 11 ARM64 versions aggresively park (offline) CPU cores and often place
+        // all our threads onto the first 4 cores which results in terrible performance with
+        // n_threads > 4
+        THREAD_POWER_THROTTLING_STATE p;
+        ZeroMemory(&p, sizeof(p));
+        p.Version     = THREAD_POWER_THROTTLING_CURRENT_VERSION;
+        p.ControlMask = THREAD_POWER_THROTTLING_EXECUTION_SPEED;
+        p.StateMask   = 0;
+
+        if (!SetThreadInformation(GetCurrentThread(), ThreadPowerThrottling, &p, sizeof(p))) {
```

Comment:
Since this function is only available on Windows 8+, it would be nice to add a check here for the Windows version. `#if _WIN32_WINNT >= 0x0602` should do it.